Sources/CRC/CRC32.swift

```diff
@@ -40,16 +40,90 @@ struct CRC32:Hashable, Sendable
         return checksum
     }
 
+    /// Returns a new checksum by hashing the provided message into the current checksum.
+    ///
+    /// This manually specialized implementation is much faster in debug mode than the
+    /// generic implementation, but exactly the same in release mode.
+    @inlinable public
+    func updated(with message:borrowing [UInt8]) -> Self
+    {
+        var checksum:Self = self
+        checksum.update(with: message)
+        return checksum
+    }
+
     /// Updates the checksum by hashing the provided message into the existing checksum.
     @inlinable public mutating
     func update(with message:borrowing some Sequence<UInt8>)
     {
         self.checksum = ~message.reduce(~self.checksum)
         {
             (state:UInt32, byte:UInt8) in
-            Self.table[Int.init(UInt8.init(truncatingIfNeeded: state) ^ byte)] ^ state >> 8
+            let indexByte:UInt8 = UInt8.init(truncatingIfNeeded: state) ^ byte
```

Comment:
since the performance-sensitive caller is going through the `[UInt8]` overload, do we actually need this?